scripts/yse/yse.py

```diff
@@ -0,0 +1,264 @@
+                  Farias, Diego and
+                  Gao, Hua and
+                  Gomez, Sebastian and
+                  Grayling, Matthew and
+                  Grillo, Claudia and
+                  Hayes, Erin E. and
+                  Hung, Tiara and
+                  Izzo, Luca and
+                  Khetan, Nandita and
+                  Kolborg, Anne Noer and
+                  Law-Smith, Jamie A. P. and
+                  LeBaron, Natalie and
+                  Lin, Chien C. and
+                  Luo, Yufeng and
+                  Magnier, Eugene A. and
+                  Matthews, David and
+                  Mockler, Brenna and
+                  O'Grady, Anna J. G. and
+                  Pan, Yen-Chen and
+                  Politsch, Collin A. and
+                  Raimundo, Sandra I. and
+                  Rest, Armin and
+                  Ridden-Harper, Ryan and
+                  Sarangi, Arkaprabha and
+                  Schrøder, Sophie L. and
+                  Smartt, Stephen J. and
+                  Terreran, Giacomo and
+                  Thorp, Stephen and
+                  Vazquez, Jason and
+                  Wainscoat, Richard and
+                  Wang, Qinan and
+                  Wasserman, Amanda R. and
+                  Yadavalli, S. Karthik and
+                  Yarza, Ricardo and
+                  Zenati, Yossef},
+  title        = {{The Young Supernova Experiment Data Release 1 (YSE 
+                   DR1) Light Curves}},
+  month        = nov,
+  year         = 2022,
+  publisher    = {Zenodo},
+  version      = {1.0.0},
+  doi          = {10.5281/zenodo.7317476},
+  url          = {https://doi.org/10.5281/zenodo.7317476}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+Time-series dataset from the Young Supernova Experiment Data Release 1 (YSE DR1).
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = "https://zenodo.org/records/7317476"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = "CC BY 4.0"
+
+_VERSION = "0.0.1"
+
+_STR_FEATURES = [
+    "object_id",
+    "spec_class",
+]
+
+_FLOAT_FEATURES = [
+    "ra", 
+    "dec", 
+    "redshift",
+    "host_log_mass"
+]
+
+
+class YSEDR1(datasets.GeneratorBasedBuilder):
+    """"""
+
+    VERSION = _VERSION
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="yse_dr1",
+            version=VERSION,
+            data_files=DataFilesPatternsDict.from_patterns({"train": ["./healpix=*/*.hdf5"]}), # This seems fairly inflexible. Probably a massive failure point.
+            description="Light curves from YSE DR1",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "yse_dr1"
+
+    @classmethod
+    def _info(self):
+        """Defines the features available in this dataset."""
+        # Starting with all features common to light curve datasets
+        features = {
```

Comment:
not sure how others feel about organizing these lightcurve features into a `lightcurve` key (ben mentioned this in https://astropile.slack.com/archives/C06R7FF78RH/p1713287345486819), but this is what i did for plasticc